### ISSUE TITLE
Notifications permission is not requested

### ISSUE THREAD

**Issue by juanlaube** [toggl/mobileapp#6802]
## 🐛 Bug Report
We used to ask for the notifications permission in the calendar onboarding, but that's gone now and we didn't move the permissions request to another place.

## Context

Build/Branch: iOS 2.8
Phone OS: iOS
Phone model:

## Expected behavior
At some point, we should ask for the notifications permission, otherwise the calendar notifications won't show up.

## To reproduce
Try to enable the calendar notifications, I dare you.

## Suggestions
I don't know what's the best place for this because we don't have an onboarding now, perhaps showing something like this in the calendar view?

🎨 
![IMG_D4AD1F29EF95-1](https://user-images.githubusercontent.com/8163330/70710265-8b519e00-1cbd-11ea-9023-5c37d3de290c.jpeg)

**Comment by juanlaube** [comment]
cc @maritoggl because this needs design (although I think my proposal is awesome and I'm very humble 😊)

**Comment by maritoggl** [comment]
Linking a related issue here as well as I'd rework the permission flow in general (currently it's broken even if one would like to grant permissions manually) https://github.com/toggl/mobileapp/issues/6797. 

Notification permission is somewhat different from the calendar permission that I'd ask right when the user first opens the calendar view (as proposed here https://github.com/toggl/mobileapp/issues/6361). Smart alerts sort of have a prerequisite that the calendar permissions are already granted and that some calendars are linked as well - otherwise there's nothing to alert users of. For that reason, I'm thinking that perhaps instead of asking the user for the notification permissions in the calendar view itself, we could "advertise" the smart alerts feature to users after they have had their calendars enabled for some period of time. Perhaps for that we could use a module in the main timer screen - the same section we currently use for asking for ratings and sort of plan to use for marketing messages as well (that include promoting various features we have). And that's when we'd need to ask for permission to notify as well. 

Long story short, the issue you've brought out is valid, but I'd first prioritize fixing the manual flow whereby permissions can still be granted via settings first. Once that's done, I'd come back to figuring out a way how to "onboard" more users to use the smart alerts feature as well and that will require us to decide where is the right place and when is the right time to proactively request the permission to send notifications.
